packages/mui-system/src/styleFunctionSx/styleFunctionSx.test.js

```diff
@@ -441,4 +441,21 @@ describe('styleFunctionSx', () => {
       letterSpacing: 'inherit',
     });
   });
+
+  it('resolves theme typography properties', () => {
+    const result = styleFunctionSx({
+      theme: { typography: createTypography({}, {}) },
+      sx: {
+        fontFamily: 'default',
+        fontWeight: 'fontWeightMedium',
+        fontSize: 'fontSize',
```

Comment:
Now I understood why [this](https://github.com/mui/material-ui/pull/33621/files#r1135213852) was added. Just the syntax looks a bit weird.

Reply:
We are going to drop that one in v6, that was the initial plan. We will support 'default', or 'light', 'medium' etc. We just cannot do it during v5's lifetime.